src/modules/cpu/kernel/resample.hpp

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+inline Rpp64f Hann(Rpp64f x) {
+    return 0.5 * (1 + std::cos(x * M_PI));
+}
+
+struct ResamplingWindow {
+    inline void input_range(Rpp32f x, Rpp32s *loc0, Rpp32s *loc1) {
+        Rpp32s xc = ceilf(x);
+        *loc0 = xc - lobes;
+        *loc1 = xc + lobes;
+    }
+
+    inline Rpp32f operator()(Rpp32f x) {
```

Comment:
Instead of using the operator overloading (Rpp32f x), operator()(__m128 x) {
please check if by using a seperate functions inside the class will give better performance

Reply:
This change doesn't seem to improve performance its either same of slightly worse, checked this with 128 audio files over 10k iterations.